ext/standard/string.c

```diff
@@ -4006,7 +4007,9 @@ static zend_long php_str_replace_in_subject(zval *search, zval *replace, zval *s
 				/* Get current entry */
 				if ((replace_entry = zend_hash_get_current_data_ex(Z_ARRVAL_P(replace), &pos)) != NULL) {
 					/* Make sure we're dealing with strings. */
-					convert_to_string_ex(replace_entry);
+					ZVAL_DEREF(replace_entry);  // Deal with possibly shared zvals
```

Comment:
The array to string conversion case is already covered by the other tests and behavior is not affected by this PR.